specs/WinRT/WinRTAPIContracts.md

```diff
@@ -0,0 +1,80 @@
+# Windows App SDK API Contracts
+
+## Summary
+Customers trust that Microsoft products are sound. As Windows App SDK evolves and more developers
+use Windows App SDK in their own projects Windows App SDK needs a way, at build time and run time,
+to allow apps to light up API features and to gracefully degrade. This can be achieved by looking
+at the deltas between APIs.
+
+In order to allow apps to depend on API versions Windows App SDK will add contracts to all their
+WinRT APIs.
+
+
+## Architecture Decisions
+1. Contract boundaries
+   1. One contract per feature/component/sub-system
+   2. Contracts will not span across a transport package/repo.
+3. Contract numbering.
+Contracts versions are one number, starting with 1 and is incremented for any interface
+change. This version does not correlate with the Windows App SDK version.
+4. Enforcement
+The build system has gates that enforce contract versioning to verify that contract is present,
+ up to date, and there are no conflicts between APIs.
+5. Contract Migration.
+This spec does not deal with contract migration. A separate spec will be available for contract
+migration if, and when, it is needed.
+6. Internal contracts
+All WinRT internal APIs have a contract, and the version is 1.  The version is never updated.
+
```

Comment:
I don't see a "Contract Naming" item in the list. Don't all contracts need to end with `...Contract`? e.g. `BlahBlahContract`